projects/kit/directives/highlight/highlight.component.ts

```diff
@@ -0,0 +1,21 @@
+import {ChangeDetectionStrategy, Component, HostBinding} from '@angular/core';
+
+@Component({
+    selector: 'tui-highlight',
+    template: '',
+    styleUrls: ['./highlight.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiHighlightComponent {
+    @HostBinding('style.left.px')
+    left = NaN;
+
+    @HostBinding('style.top.px')
+    top = NaN;
+
+    @HostBinding('style.width.px')
+    width = NaN;
+
+    @HostBinding('style.height.px')
+    height = NaN;
+}
```

Comment:
Can we use native `mark` (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/mark) instead of angular component?

Reply:
This is interesting, but using mark implies modifying an existing DOM tree, which can lead to unexpected consequences. Mark is good for static text selection, but not for dynamic. In my opinion.

Reply:
okey